lib/HttpClient/CurlClient.php

```diff
@@ -365,51 +364,41 @@ private function shouldRetry($errno, $isPost, $rcode, $rbody, $numRetries)
             return true;
         }
 
-        // 409 Conflict
-        if ($rcode === 409) {
-            return true;
+        // The API may ask us not to retry (eg; if doing so would be a no-op)
+        // or advise us to retry (eg; in cases of lock timeouts); we defer to that.
+        if ($rheaders['stripe-should-retry'] === 'false') {
```

Comment:
This works, but causes a warning if the key doesn't exist.

```suggestion
        if (isset($rheaders['stripe-should-retry']) && $rheaders['stripe-should-retry'] === 'false') {
```